lib/APMSpan.js

```diff
@@ -21,31 +21,41 @@ import { APMSpanContext } from './APMSpanContext';
 class APMSpan {
     constructor(tracer, fields) {
         this._tracer = tracer;
-        this._spanContext = APMSpan._createSpanContext(fields.childOf);
+        if (fields.childOf && fields.childOf instanceof APMSpan) {
+            fields.childOf = fields.childOf.context();
+        }
+
+        this._initSpanContext(fields.childOf);
 
         this._operationName = fields.operationName;
         this._tags = fields.tags || {};
         this._startMillis = fields.startTime || nowMillis();
         this._endMillis = null;
         this._logs = [];
-        this._sampled = tracer.getSampler().isSampled(this);
 
         this._finished = false;
     }
 
-    static _createSpanContext(parentSpanOrContext) {
+    _initSpanContext(parentSpanContext) {
         const id = generateSpanId();
```

Comment:
I believe this id is the one used for the 'interaction id' exchanged between client and server ? If so, then if two separate service invocations occurred within the scope of a single span, they would both have the same id - which would be incorrect, as they are supposed to be unique per interaction.

Although the app developer should use a child span per service invocation, there is nothing that dictates this. However if the id is generated at the point of injection, and associated with the Producer node as a 'interaction' based correlation id, in the mentioned scenario we would just end up with a Producer node having two correlation ids, which is supported by the model.